Sming/Libraries/Servo/ServoChannel.h

```diff
@@ -9,83 +9,90 @@
  *  @{
 */
 
-#ifndef LIB_SERVOCHANNEL_H_
-#define LIB_SERVOCHANNEL_H_
+#pragma once
 
 #include <SmingCore.h>
 
-#define DEFAULTMAXVALUE 2300	///< default max value for output in us
-#define DEFAULTMINVALUE 700		///< default min value for output in us
-
+#define DEFAULTMAXVALUE 2300 ///< default max value for output in us
+#define DEFAULTMINVALUE 700  ///< default min value for output in us
 
 class ServoChannel
 {
 public:
-    /** @brief  Instantiate a servo channel object\n\r
+	/** @brief  Instantiate a servo channel object\n\r
      * has to be attached to a pin before using
      */
 	ServoChannel();
 	virtual ~ServoChannel();
 
-    /** @brief  attach servo channel to a unused pin
+	/** @brief  attach servo channel to a unused pin
      *  @param  pin GPIO to use\n\r
      * pin will be set to output
      *  @retval bool True on success
      */
 	bool attach(uint8 pin);
-    /** @brief  detaches servo channel
+	/** @brief  detaches servo channel
      * pin will be set to input
      *  @retval bool True on success
      */
 	bool detach();
 
-    /** @brief  set value in servo degree\n\r
+	/** @brief  set value in servo degree\n\r
      * input value is mapped between min and max value
      *  @param  value is limited between -90 and 90 degrees
      *  @retval bool True on success
      */
 	bool setDegree(int8 value);
 
-    /** @brief  set max value for output timing
+	/** @brief  set max value for output timing
      *  @param  value time in us for maximum output of this channel\n\r
      *  has to be larger than min value
      *  @retval bool True on success
      */
 	bool setMaxValue(uint32 maxValue = DEFAULTMAXVALUE);
 
-    /** @brief  set min value for output timing
+	/** @brief  set min value for output timing
      *  @param  value time in us for minimum output of this channel\n\r
      *  has to be smaller than max value
      *  @retval bool True on success
      */
 	bool setMinValue(uint32 minValue = DEFAULTMINVALUE);
 
-    /** @brief  simple getter
+	/** @brief  simple getter
      *  @retval uint32 max time in us
      */
-	uint32 getMaxValue() const { return maxValue; }
+	uint32 getMaxValue() const
+	{
+		return maxValue;
+	}
 
-    /** @brief  simple getter
+	/** @brief  simple getter
      *  @retval uint32 min time in us
      */
-	uint32 getMinValue() const { return minValue; }
+	uint32 getMinValue() const
+	{
+		return minValue;
+	}
 
-    /** @brief  simple getter
+	/** @brief  simple getter
      *  @retval uint32 actual value in us
      */
 	uint32 getValue() const;
 
-    /** @brief  set value for output
+	/** @brief  set value for output
      *  @param  value time in us for actual output of this channel\n\r
      *  has to be smaller than max-min value
      *  @retval bool True on success
      */
 	bool setValue(uint32 value);
 
```

Comment:
I find this approach highly confusing. The logical expectation is that `minValue <= value <= maxValue` (a).
In reality it's `0 <= value <= (maxValue - minValue)` (b).

My preference would be to amend the API to define value as for (a), but that will break things.

The other approach would be to rename `minValue` and `maxValue`.

Thoughts?

Reply:
> My preference would be to amend the API

Go for it. As far as I can see the impact will be relatively small within Sming. That will require good documentation too.

Reply:
OK, I've amended the API and revised the library so interrupts remain enabled. Scope output solid and servos work fine.